.github/workflows/ci.yml

```diff
@@ -24,89 +19,54 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
-
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python-version }}
-
     - name: Install uv
-      run: |
-        curl -LsSf https://astral.sh/uv/install.sh | sh
-        echo "$HOME/.cargo/bin" >> $GITHUB_PATH
-
-    - name: Setup virtual environment and install dependencies
-      run: |
-        uv venv
-        source .venv/bin/activate
-        uv pip install -e ".[dev]"
+      uses: astral-sh/setup-uv@v3
+      with:
+        version: "0.4.x"
 
     - name: Run tests
       run: |
-        source .venv/bin/activate
         uv run pytest tests/ -v
+      env:
+        UV_PYTHON: ${{ matrix.python-version }}
 
   lint:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-
-    - name: Set up Python
-      uses: actions/setup-python@v5
-      with:
-        python-version: "3.12"
-
     - name: Install uv
-      run: |
-        curl -LsSf https://astral.sh/uv/install.sh | sh
-        echo "$HOME/.cargo/bin" >> $GITHUB_PATH
-
-    - name: Setup virtual environment and install dependencies
-      run: |
-        uv venv
-        source .venv/bin/activate
-        uv pip install ruff black
+      uses: astral-sh/setup-uv@v3
+      with:
+        version: "0.4.x"
 
     - name: Lint with ruff
       run: |
-        source .venv/bin/activate
-        ruff check .
+        uvx --python=3.12 ruff check .
 
     - name: Check formatting with black
       run: |
-        source .venv/bin/activate
-        black --check .
+        uvx --python=3.12 black --check .
 
   publish:
     name: pypi-publish
     needs: [test, lint]
     runs-on: ubuntu-latest
-    environment:
-      name: release
-      url: https://pypi.org/p/grid_df
```

Comment:
Are these configured for the repo on PyPI?

Reply:
Here's a workflow of mine for a recent project: https://github.com/manzt/juv/blob/main/.github/workflows/release.yml

On PyPI, I just have the repo and workflow name configured (no enviroment):

<img width="711" alt="image" src="https://github.com/user-attachments/assets/8b5c85bc-4cc0-4297-9dbb-22784ba2abde">